src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerFileReference.cs

```diff
@@ -111,17 +111,30 @@ public override int GetHashCode()
 
         public override ImmutableArray<DiagnosticAnalyzer> GetAnalyzersForAllLanguages()
         {
-            return _diagnosticAnalyzers.GetExtensionsForAllLanguages();
+            // This API returns duplicates of analyzers that support multiple languages.
+            // We explicitly retain this behaviour to ensure back compat
+            return _diagnosticAnalyzers.GetExtensionsForAllLanguages(includeDuplicates: true);
```

Comment:
📝I'm not sure if it was ever intentional to return duplicate analyzer references, but I've ensured we keep the previous behavior.

Reply:
Should we just break this? I admit I can't think of anything that would care, and honestly more that might be broken. :-)

Reply:
Yeah, I have no idea. It *is* a public API though, and it's been around for at least 6 years according to blame.